src/components/cursor.js

```diff
@@ -33,60 +38,118 @@ module.exports.Component = registerComponent('cursor', {
 
-    // Wait for canvas to load.
-    if (!canvas) {
-      el.sceneEl.addEventListener('render-target-loaded', bind(this.init, this));
-      return;
-    }
-
     // Bind methods.
-    this.onMouseDown = bind(this.onMouseDown, this);
-    this.onMouseUp = bind(this.onMouseUp, this);
+    this.onCursorDown = bind(this.onCursorDown, this);
+    this.onCursorUp = bind(this.onCursorUp, this);
     this.onIntersection = bind(this.onIntersection, this);
     this.onIntersectionCleared = bind(this.onIntersectionCleared, this);
+  },
 
-    // Attach event listeners.
-    canvas.addEventListener('mousedown', this.onMouseDown);
-    canvas.addEventListener('mouseup', this.onMouseUp);
-    el.addEventListener('raycaster-intersection', this.onIntersection);
-    el.addEventListener('raycaster-intersection-cleared', this.onIntersectionCleared);
+  update: function (oldData) {
+    var data = this.data;
+    var el = this.el;
+    var raycaster = el.components.raycaster.raycaster;
+    var rayLength;
+
+    // Create laser beam.
+    if (data.laserEnabled) {
+      rayLength = raycaster.far === Infinity ? 1000 : raycaster.far;
+      el.setAttribute('line', {
+        color: data.laserColor,
+        start: raycaster.ray.origin,
+        end: endVec3.copy(raycaster.ray.origin).add(
+          directionVec3.copy(raycaster.ray.direction).multiplyScalar(rayLength)
+        ),
+        width: data.laserWidth
+      });
+    }
+
+    if (data.laserEnabled && oldData.laserEnabled) { el.removeAttribute('line'); }
+  },
+
+  play: function () {
+    this.addEventListeners();
+  },
+
+  pause: function () {
+    this.removeEventListeners();
   },
 
   remove: function () {
+    var data = this.data;
     var el = this.el;
-    var canvas = el.sceneEl.canvas;
-
     el.removeState(STATES.HOVERING);
     el.removeState(STATES.FUSING);
-    el.removeEventListener('raycaster-intersection', this.onIntersection);
-    el.removeEventListener('raycaster-intersection-cleared', this.onIntersectionCleared);
-
     clearTimeout(this.fuseTimeout);
-
     if (this.intersectedEl) { this.intersectedEl.removeState(STATES.HOVERED); }
+    this.removeEventListeners();
+    if (data.laserEnabled) { el.removeAttribute('line'); }
+  },
 
+  addEventListeners: function () {
+    var canvas;
+    var data = this.data;
+    var el = this.el;
+    var self = this;
+
+    canvas = el.sceneEl.canvas;
     if (canvas) {
-      canvas.removeEventListener('mousedown', this.onMouseDown);
-      canvas.removeEventListener('mouseup', this.onMouseUp);
+      canvas.addEventListener('mousedown', this.onCursorDown);
+      canvas.addEventListener('mouseup', this.onCursorUp);
     }
+
+    data.downEvents.forEach(function (downEvent) {
+      el.addEventListener(downEvent, self.onCursorDown);
+    });
+    data.upEvents.forEach(function (upEvent) {
+      el.addEventListener(upEvent, self.onCursorUp);
+    });
+    el.addEventListener('raycaster-intersection', this.onIntersection);
```

Comment:
do you think it'll be confusing for folks if the event is named `cursor` but the event names are prefixed with `raycaster-`?